src/sentry/services/hybrid_cloud/integration/__init__.py

```diff
@@ -0,0 +1,194 @@
+            external_id=integration.external_id,
+            name=integration.name,
+            metadata=integration.metadata,
+            status=integration.status,
+        )
+
+    def _serialize_organization_integration(
+        self, oi: OrganizationIntegration
+    ) -> APIOrganizationIntegration:
+        return APIOrganizationIntegration(
+            id=oi.id,
+            default_auth_id=oi.default_auth_id,
+            organization_id=oi.organization_id,
+            integration_id=oi.integration_id,
+            config=oi.config,
+            status=oi.status,
+        )
+
+    @abstractmethod
+    def get_many(
+        self, *, integration_ids: Iterable[int] | None = None, organization_id: int | None = None
+    ) -> List[APIIntegration]:
+        """
+        Returns a list of APIIntegrations filtered either by a list of integration ids, or a single organization id
+        """
+        pass
+
+    @abstractmethod
+    def page_integration_ids(
+        self,
+        *,
+        provider_keys: List[str],
+        organization_id: int,
+        args: ApiPaginationArgs,
+    ) -> ApiPaginationResult:
+        pass
+
+    @abstractmethod
+    def get_integration(
+        self,
+        integration_id: int | None = None,
+        provider: str | None = None,
+        external_id: str | None = None,
+    ) -> APIIntegration | None:
+        """
+        Returns an APIIntegration using either the id or a combination of the provider and external_id
+        """
+        pass
+
+    @abstractmethod
+    def get_organization_integrations(
+        self, integration_id: int
+    ) -> List[APIOrganizationIntegration]:
+        """
+        Returns all APIOrganizationIntegrations associated with a given integration.
+        """
+        pass
+
+    def get_organization_integration(
+        self, integration_id: int, organization_id: int
+    ) -> APIOrganizationIntegration | None:
+        """
+        Returns an APIOrganizationIntegration from the integration and organization ids.
+        """
+        all_ois = self.get_organization_integrations(integration_id=integration_id)
+        organization_integration = next(
+            (oi for oi in all_ois if oi.organization_id == organization_id), None
+        )
+
+        return (
+            self._serialize_organization_integration(organization_integration)
+            if organization_integration
+            else None
+        )
+
+    def update_config(
+        self, org_integration_id: int, config: Mapping[str, Any], should_clear: bool = False
+    ) -> APIOrganizationIntegration | None:
+        """
+        Returns an APIOrganizationIntegration if the associated org_integration.config value
+        was successfully updated, otherwise returns None.
+        If should_clear is True, runs dict.clear() first, otherwise just uses dict.update()
+        """
+
+    def get_installation(
+        self, integration_id: int, organization_id: int, **kwargs: Mapping[str, Any]
```

Comment:
This part concerns me, in general having untyped mapping in our protocol could lead to some future grief.  Are we using this?  If we're not using it now, I'd suggest just dropping it for now.